qemu/tests/discard_for_raw_block_target.py

```diff
@@ -0,0 +1,56 @@
+import logging
+import os
+
+from provider.qemu_img_utils import strace
+
+from avocado import fail_on
+from avocado.utils import process
+
+from virttest import data_dir
+from virttest.qemu_storage import QemuImg
+
+
+def run(test, params, env):
+    """
+    qemu-img supports 'discard' for raw block target images.
+
+    1. Create source image via dd with all zero.
+    2. Modprobe a 1G scsi_debug disk with writesame_16 mode.
+    3. Trace the system calls while converting the zero image to the
+        scsi_debug block device, then check whether 'fallocate' system call
+        works in the log.
+    :param test: Qemu test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def check_output(strace_output, match_str):
+        """Check whether the value is good in the output file."""
+        logging.debug("Check the output file '%s'" % strace_output)
+        with open(strace_output) as fd:
+            if match_str not in fd.read():
+                test.fail("The system call is not right, "
+                          "check '%s' for more details" % strace_output)
+
+    src_image = params["images"]
+    source = QemuImg(params.object_params(src_image),
+                     data_dir.get_data_dir(), src_image)
+    src_filename = source.image_filename
+    strace_events = params["strace_event"].split()
+    strace_output_file = os.path.join(test.debugdir,
+                                      "convert_to_block.log")
+
+    os.system("dd if=/dev/zero of=%s bs=1M count=1024" % src_filename)
```

Comment:
Maybe this command can be written in the .cfg file.
like :dd_cmd = "dd if=/dev/urandom of=%s oflag=direct bs=1M count=10"(on .cfg)
then :dd_cmd = params["dd_cmd"] (on .py)
What do you think？

Reply:
I think it could be a little hard to understand `% src_filename` field in the py file. You need to go to the cfg file to see what this field mean.So I prefer this in the py file then. Thanks.